### ISSUE TITLE
Old component option syntax breaking video js.

### ISSUE THREAD

**Issue by billmalarky** [videojs/video.js#3035]
[I was asked to create a new issue](https://github.com/videojs/video.js/issues/953#issuecomment-174408956) since the legacy component option syntax seems to be breaking video js in some way. Here is what is breaking for me with video js version 5.4.4.

```javascript
var options= {
    children: {
        controlBar: {
            children: {
                fullscreenToggle: false
            }
        }
    }
};
                
//Load video JS
videojs('videoTagId', options, function(){
    //whatever
});
```

**Comment by gkatsev** [comment]
Thanks. Are you seeing any errors in the console?

**Comment by billmalarky** [comment]
No errors at all, which was strange. The video fails to load, and it also doesn't fallback to native html5 video either, so video js is definitely firing.

**Comment by hartman** [comment]
Testcase:
https://jsfiddle.net/TheDJ/wuvhkuba/2/

**Comment by gkatsev** [comment]
Interesting, thanks for the testcase @hartman.
Now comes the hard part in figuring out what the problem is and whether we are going to support this older syntax :P

**Comment by billmalarky** [comment]
Here's my opinion on it that might help you make the best decision for your vision of the project:

Did the breaking change occur in the last major release? And after diagnosing the cause are there any other components breaking as a result of this bug or is the bug constrained to the logic that handles the old syntax? 

If that is the case, and this is the first time it has been brought to your attention it may warrant simply updating the major release breaking changes documentation. Especially since the issue clearly isn't impacting many users if this is the first time you have been made aware of it 4 months after releasing 5.0.0.

If the breaking change occurred in a minor release, the old syntax would need to be supported in a new patched version in order to follow semvar.

http://semver.org/#what-do-i-do-if-i-accidentally-release-a-backwards-incompatible-change-as-a-minor-version

Regardless the new syntax is greatly preferable for new users like myself.

**Comment by gkatsev** [comment]
Yeah, the breakage was definitely done as part of a major release, whether intentionally or unintentionally.
I'm going to re-open this so that someone (probably me) will remember to investigate why this breakage occurred to make sure there aren't any tangential bugs.
Thanks for your comments.

**Comment by billmalarky** [comment]
Oops I accidentally clicked "close and comment" rather than comment. I'm one of the (very) few more used to bitbucket's UI than github I'm afraid.

**Comment by gkatsev** [comment]
Heh, oops.
